crates/dids/src/method/jwk.rs

```diff
@@ -0,0 +1,120 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError};
+use crate::resolver::DidResolutionResult;
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_jwk::DIDJWK as SpruceDidJwkMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:jwk DID
+pub struct DidJwk {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
```

Comment:
When creating an instance of `DidJwk`, we're going to add validation in kt. This validation will check that all the required keys for managing the given `did_uri` already exist within the `key_manager`. The bulk of this logic exists in https://github.com/TBD54566975/web5-kt/blob/dbc1df5aba9206a35791e6a20f6aa26acddc8e7f/dids/src/main/kotlin/web5/sdk/dids/DidMethod.kt#L193C1-L193C1

Would it make sense to do something similar here? No idea how you could do so in rust.

Reply:
`DidJwk` is a public struct. IIUC, consumers can instantiate is by doing something like the example below. 

```rs
let my_did_jwk = DidJwk {
    uri: String::from("did:jwk:example"),
    key_manager: Arc::clone(keyManager), // keyManager does not have the keys for `did:jwk:example`!!
};
```
Is my understanding correct? 

If so, then my suggestion is to either: a) prevent consumers from doing such a construction (since you cannot use `my_did_jwk` to sign stuff if it doesn't have the keys) or b) validate that the keys exist inside the `keyManager` upon instantiation. Is something like that reasonable?

Reply:
Your understanding isn't quite correct here. That manner of construction is only supported _within_ the `dids` crate due to Rust's access control.

Here's an example of someone trying to consume the `dids` crate and create a new `DidJwk` in the manner you suggest:

![Screenshot 2023-12-04 at 11 30 32 AM](https://github.com/TBD54566975/web5-rs/assets/88001738/ef51a2f3-18c8-40b3-aa90-df4f72e4970e)

This will not compile, as `uri` and `key_manager` are not public. It will result in the following compilation error:
```
❯ cargo build -p example
   Compiling example v0.1.0 (/Users/amika/Development/rust/web5-rs/crates/show-andres-hes-wrong)
error[E0451]: field `uri` of struct `DidJwk` is private
 --> crates/show-andres-hes-wrong/src/lib.rs:7:9
  |
7 |         uri: "did:jwk:1234".to_string(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ private field

error[E0451]: field `key_manager` of struct `DidJwk` is private
 --> crates/show-andres-hes-wrong/src/lib.rs:8:9
  |
8 |         key_manager: Arc::new(LocalKeyManager::new_in_memory()),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ private field

For more information about this error, try `rustc --explain E0451`.
error: could not compile `example` (lib) due to 2 previous errors
```

If we want external users of the `dids` crate to create arbitrary `DidJwk` objects, we would either need to do one of the following:
1. Make `uri` and `key_manager` public on `DidJwk`
2. Make a public constructor that allows the user to provide their own `uri` and `key_manager`

I would prefer option 2, as we can encapsulate validation logic within a constructor.

Reply:
Ohhh thank you so much. That makes a ton of sense and I appreciate the explanation!